drivers/adc/adc_context.h

```diff
@@ -165,6 +174,10 @@ static inline int adc_context_wait_for_completion(struct adc_context *ctx)
 
 static inline void adc_context_complete(struct adc_context *ctx, int status)
 {
+#ifdef ADC_CONTEXT_ENABLE_ON_COMPLETE
+	adc_context_on_complete(ctx, status);
```

Comment:
May I ask why is it needed to add this here? It's the driver who calls `adc_context_complete()`, so if it needs to perform some additional operations at that point why cannot it do this directly? What's the benefit of doing this through the adc_context layer?

Reply:
This is because `adc_stm32_teardown_channels` needs to happen after all sampling is finished. Doing it in `adc_context_on_sampling_done` is not enough, because that can trigger another `adc_context_start_sampling`, which will then sample the ADC _after_ the channel are tore down, without reconfiguring it again.

You can find the conversation that sparked this change [here](https://github.com/zephyrproject-rtos/zephyr/pull/53534#discussion_r1063814975).

Reply:
Okay, I understand. Thanks.